--- a/PEAR/Builder.php	2011-05-14 20:43:01.000000000 +0000
+++ b/PEAR/Builder.php		2011-05-26 15:56:41.096485701 +0000
@@ -309,6 +309,8 @@ class PEAR_Builder extends PEAR_Common
         }
 
         if (!$err) {
+             print "If the command failed with 'phpize: not found' then you need to install php5-dev package";
+             print "You can do it by running 'apt-get install php5-dev' as a root user";
             return $this->raiseError("`phpize' failed");
         }
