--- ffmpeg-3.2/libavfilter/Makefile	2016-06-27 00:54:30.000000000 +0100
+++ ffmpeg313_patches/libavfilter/Makefile	2016-09-19 05:01:40.000000000 +0100
@@ -292,6 +292,7 @@
 OBJS-$(CONFIG_ZMQ_FILTER)                    += f_zmq.o
 OBJS-$(CONFIG_ZOOMPAN_FILTER)                += vf_zoompan.o
 OBJS-$(CONFIG_ZSCALE_FILTER)                 += vf_zscale.o
+OBJS-$(CONFIG_TRANSFORM_FILTER) += vf_transform.o
 
 OBJS-$(CONFIG_ALLRGB_FILTER)                 += vsrc_testsrc.o
 OBJS-$(CONFIG_ALLYUV_FILTER)                 += vsrc_testsrc.o
